app/main/util/render_utils.py

```
@@ -56,6 +58,66 @@ def create_presigned_url(file: File) -> str:
+                mat = pymupdf.Matrix(DPI / 72, DPI / 72)
+                pix = page.get_pixmap(matrix=mat)
+                img_bytes = pix.tobytes("png")
+
+                # Convert to PIL Image for thumbnail processing
+                page_image = Image.open(io.BytesIO(img_bytes))
+
+                # Create thumbnail (150x200 pixels)
+                thumbnail = page_image.copy()
+                thumbnail.thumbnail((150, 200), Image.Resampling.LANCZOS)
+
+                # Convert thumbnail to base64 data URL
+                thumbnail_buffer = io.BytesIO()
+                thumbnail.save(thumbnail_buffer, format="JPEG", quality=70)
+                thumbnail_base64 = base64.b64encode(
+                    thumbnail_buffer.getvalue()
+                ).decode()
+                thumbnail_data_url = (
+                    f"data:image/jpeg;base64,{thumbnail_base64}"
+                )
+
+                # Convert full page to base64 data URL
+                page_buffer = io.BytesIO()
+                page_image.save(page_buffer, format="JPEG", quality=75)
+                page_base64 = base64.b64encode(page_buffer.getvalue()).decode()
+                page_data_url = f"data:image/jpeg;base64,{page_base64}"
+
+                current_app.logger.debug(
+                    f"Page {page_num + 1}: thumbnail={len(thumbnail_base64)} chars, full={len(page_base64)} chars"
+                )
+
+                page_data.append(
+                    {
+                        "page_number": page_num + 1,
+                        "width": page_image.width,
+                        "height": page_image.height,
+                        "thumbnail_url": thumbnail_data_url,
+                        "page_image_url": page_data_url,
+                    }
+                )
+
+                # Clean up resources
+                page_image.close()
+                thumbnail.close()
+                pix = None
+
+            return page_data
+    except Exception as e:
```

maybe remove this, as we dont want to silently continue